std-bits/base/src/main/java/org/enso/base/cache/LRUCache.java

```diff
@@ -128,27 +119,26 @@ private File downloadResponseData(String cacheKey, Item item)
       throws IOException, ResponseTooLargeException {
     File temp = File.createTempFile("LRUCache-" + cacheKey, "");
     temp.deleteOnExit();
+    var inputStream = item.stream();
+    var outputStream = new FileOutputStream(temp);
+    boolean successful = false;
     try {
-      var inputStream = item.stream();
-      try (var outputStream = new FileOutputStream(temp)) {
-
-        // Limit the download to getMaxFileSize().
-        boolean sizeOK = Stream_Utils.limitedCopy(inputStream, outputStream, getMaxFileSize());
-        if (!sizeOK) {
-          try {
-            temp.delete();
-            outputStream.close();
-          } finally {
-            // catch block below will delete the temp file.
-            throw new ResponseTooLargeException(getMaxFileSize());
-          }
-        }
+      // Limit the download to getMaxFileSize().
+      boolean sizeOK = Stream_Utils.limitedCopy(inputStream, outputStream, getMaxFileSize());
 
+      if (sizeOK) {
+        successful = true;
         return temp;
+      } else {
+        throw new ResponseTooLargeException(getMaxFileSize());
+      }
+    } finally {
+      if (!successful) {
+        outputStream.close();
```

Comment:
Shouldn't we close the stream regardless of if we were successful or not?

Reply:
Fixed

Reply:
Github really wants me to reply to this again so here I am replying.